modules/cats/src/test/scala/pureconfig/module/cats/CatsSuite.scala

```diff
@@ -16,13 +16,17 @@ class CatsSuite extends BaseSuite with ConfigConvertChecks {
   case class NumVec(numbers: NonEmptyVector[Int])
   case class NumSet(numbers: NonEmptySet[Int])
   case class NumMap(numbers: NonEmptyMap[String, Int])
+  case class NumChain(numbers: NonEmptyChain[Int])
 
   checkReadWrite[Numbers](parseString(s"""{ numbers: [1,2,3] }""").root() → Numbers(NonEmptyList(1, List(2, 3))))
   checkReadWrite[NumVec](parseString(s"""{ numbers: [1,2,3] }""").root() → NumVec(NonEmptyVector(1, Vector(2, 3))))
   checkReadWrite[NumSet](parseString(s"""{ numbers: [1,2,3] }""").root() → NumSet(NonEmptySet(1, SortedSet(2, 3))))
   checkReadWrite[NumMap](parseString(s"""{ 
                                            numbers {"1": 1, "2": 2, "3": 3 }
                                          }""").root() → NumMap(NonEmptyMap(("1", 1), SortedMap("2" → 2, "3" → 3))))
+  // Chain seems to break referential transparency unless prepend/append order is the same
+  checkReadWrite[NumChain](parseString(s"""{ numbers: [1,2,3] }""").root() →
```

Comment:
Hi @ruippeixotog , thanks for your feedback. I was having no issues when I try defining an `Equality` for `Chain`, but for `NEC[Int]` there were type erasure issues from its underlying dependent type of `NonEmptyChainImpl.Type`, so I ended up defining an `Equality` for `NumChain` instead.

Also, the equality issue should be fixed from Cats 1.6. https://github.com/typelevel/cats/pull/2690

Also related, https://github.com/typelevel/cats/pull/522 . But it uses `Equivalence` instead of `Equality`. Perhaps the helpers in `ConfigConvertChecks` could depend on `Equivalence` instead.

Reply:
I don't really have a preference between `Equivalance` and `Equality`.

Semantically it seems to make more sense to define convert tests on equality because we are not really doing computation that can take approximate results; in the case of `Chain`, I'd consider the two chains in question equal, not equivalent. Nonetheless, I'd be ok with using `Equivalence` if it helped somehow :)

Reply:
`Equivalence` is not necessary (at least for this PR) as I've already resolved it in https://github.com/pureconfig/pureconfig/pull/445/commits/11240b001be0ec7f943c6791787d9960feff986f with an instance of `Equality[NumChain]` (`NumChain` wraps `NEC[Int]`, kind of like a `ClassTag`) instead of `Equality[NEC]`, unless the latter is preferred. (And it is probably redundant from Cats 1.6 anyway.)

That said I still think that it is better to switch over eventually, perhaps in another PR. ["Scalactic provides both Equality and Equivalence because the Any in Equality can sometimes make things painful."](https://www.artima.com/docs-scalatest-3.0.0-SNAP1/org/scalactic/Equality.html) , and in this case, together with how `NEC` is implemented, causes the type erasure. Plus you will be able to get `cats.Eq` into scalactic for free.